modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java

```diff
@@ -0,0 +1,245 @@
+ * limitations under the License.
+ */
+
+package org.wso2.am.integration.tests.apiproduct.lifecycle;
+
+import org.junit.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.clients.publisher.api.ApiException;
+import org.wso2.am.integration.clients.publisher.api.ApiResponse;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIKeyDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIProductDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.LifecycleStateDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.SubscriptionListDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.WorkflowResponseDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationKeyDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationKeyGenerateRequestDTO;
+import org.wso2.am.integration.test.impl.ApiProductTestHelper;
+import org.wso2.am.integration.test.impl.ApiTestHelper;
+import org.wso2.am.integration.test.impl.InvocationStatusCodes;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleState;
+import org.wso2.am.integration.test.utils.http.HttpRequestUtil;
+import org.wso2.am.integration.tests.api.lifecycle.APIManagerLifecycleBaseTest;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.ws.rs.core.Response;
+import javax.xml.xpath.XPathExpressionException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+public class APIProductLifecycleTest extends APIManagerLifecycleBaseTest {
+
+    private ApiProductTestHelper apiProductTestHelper;
+    private ApiTestHelper apiTestHelper;
+    private List<APIDTO> apisToBeUsed;
+    private String apiProductId;
+    private String applicationId;
+    private ApplicationKeyDTO productionAppKey;
+    private InvocationStatusCodes invocationStatusCodes = new InvocationStatusCodes();
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public APIProductLifecycleTest(TestUserMode userMode) {
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN},
```

Comment:
Shall we add the other user modes as well?

Reply:
I will test for other user modes and send it in a different PR

Reply:
Addressed in https://github.com/wso2/product-apim/pull/12116